### PR TITLE
add links to the opt out message

### PR DESCRIPTION
**Pull request by camallen**
use markdown to render the optOut string correctly with a link to study. ~Also test the default translation fall back works~

Staging branch URL:

Fixes # .

Describe your changes.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/classifier/components/Intervention/Intervention.spec.js [hunk 1]
```diff
@@ -17,20 +17,31 @@ describe('Intervention', function () {
...
   it('should render', function () {
     expect(wrapper).to.be.ok;
   });
   it('should show a notification message', () => {
     const newlineMsg = intervention.message + "\n"
-    expect(wrapper.find(Markdown).text()).to.equal(newlineMsg);
+    expect(wrapper.find(Markdown).first().text()).to.equal(newlineMsg);
+  });
+  it('should show an opt-out message', () => {
+    const optOutHTML = '<div class="markdown "><p>I do not want to take part in this messaging ' +
```
Comment (camallen) [inline app/classifier/components/Intervention/Intervention.spec.js]: I'm not super happy testing the HTML here, but i wasn't sure what to do, i could check for a link `<a href=` tag vs the component html but lose some granularity over the test. 

Also for some reason there is a new line before the last div 🤷‍♂

File: app/classifier/components/Intervention/Intervention.spec.js [hunk 2]
```diff
@@ -17,20 +17,31 @@ describe('Intervention', function () {
...
 
-  before(function () {
-    wrapper = mount(
-      <Intervention
-        intervention={intervention}
-        onUnmount={onUnmount}
-        user={user}
-      />);
-  });
+  let wrapper = mount(
```
Comment (eatyourgreens) [inline app/classifier/components/Intervention/Intervention.spec.js]: Why did you delete the `before` hook?

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply app/classifier/components/Intervention/Intervention.spec.js]: Yeah, this is a really brittle test.

File: app/classifier/components/Intervention/Intervention.spec.js [hunk 3]
```diff
@@ -17,20 +17,31 @@ describe('Intervention', function () {
...
+    expect(wrapper.find(Markdown).first().text()).to.equal(newlineMsg);
+  });
+  it('should show an opt-out message', () => {
+    const optOutHTML = '<div class="markdown "><p>I do not want to take part in this messaging ' +
+      '<a href="https://docs.google.com/document/d/1gLyN6Dgff8dOCOC88f47OD6TtFrfSJltsLgJMKkYMso/preview" ' +
+      'target="_blank" ref="noopener nofollow">study</a>. Do not show me further messages.</p>' +
+      '\n' + '</div>'
+    expect(wrapper.find(Markdown).last().html()).to.equal(optOutHTML);
+  });
+  it('should show a fallback to english opt-out message', () => {
```
Comment (eatyourgreens) [inline app/classifier/components/Intervention/Intervention.spec.js]: This is testing counterpart, not the component.

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 1 above)
Comment (srallen) [reply app/classifier/components/Intervention/Intervention.spec.js]: Could try [`contains`](https://airbnb.io/enzyme/docs/api/ShallowWrapper/contains.html) instead? I sometimes find that a bit more readable than the html render method for shallow.

Comment (eatyourgreens) [reply app/classifier/components/Intervention/Intervention.spec.js]: Examining the props of the Markdown component might allow you to test the markdown string that’s passed into it as `props.children`.

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 3 above)
Comment (camallen) [reply app/classifier/components/Intervention/Intervention.spec.js]: The test was on the wrapper component output, which relies on counterpart. Sure i had to modify counterpart in setup...anyway i can't reliably get the test to fail as counterpart seems to have fallbackLocales setup by default. 

I couldn't see where this config was being set, default? So I'm going to assume that this is normal, expected behaviour and remove the test.

```
{ locale: 'en',
     interpolate: true,
     fallbackLocales: [ 'en' ],
     scope: null,
     translations: 
      { en: [Object],
        cs: [Object],
        it: [Object],
        es: [Object],
        nl: [Object],
        pt: [Object] },
     interpolations: {},
     normalizedKeys: { '.': [Object] },
     separator: '.',
     keepTrailingDot: false,
     keyTransformer: [Function: keyTransformer],
     generateMissingEntry: [Function: generateMissingEntry] },
```

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 1 above)
Comment (camallen) [reply app/classifier/components/Intervention/Intervention.spec.js]: Thanks that is a much nicer solution :)

File: app/classifier/components/Intervention/Intervention.spec.js [hunk 4]
```diff
@@ -30,7 +30,14 @@ describe('Intervention', function () {
   });
   it('should show a notification message', () => {
     const newlineMsg = intervention.message + "\n"
-    expect(wrapper.find(Markdown).text()).to.equal(newlineMsg);
+    expect(wrapper.find(Markdown).first().text()).to.equal(newlineMsg);
+  });
+  it('should show an opt-out message', () => {
+    const expectedMarkdown = 'I do not want to take part in this messaging [study]' +
```
Comment (eatyourgreens) [inline app/classifier/components/Intervention/Intervention.spec.js]: You can also get this string by loading the English language file, but that might be a lot of work for a single string.

Comment (camallen) [reply app/classifier/components/Intervention/Intervention.spec.js]: I really want to make sure that the test has the study link setup, if we use the translation strings themselves and someone removes the link in the translation string the tests will pass but without the correct study link.

Comment (eatyourgreens) [reply app/classifier/components/Intervention/Intervention.spec.js]: That makes sense, but we're now maintaining the text in two places, which is slightly annoying.

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 3 above)
Comment (eatyourgreens) [reply app/classifier/components/Intervention/Intervention.spec.js]: From memory, [counterpart falls back to English by default](https://www.npmjs.com/package/counterpart#locales). We should probably update that to set the default locale to the project primary language when a project loads, if we don't do that already. It just so happens that counterpart's default is the same as Panoptes' default, so it magically works.

Comment (eatyourgreens) [reply app/classifier/components/Intervention/Intervention.spec.js]: Ignore that. We set the fallback when the JS first loads.
https://github.com/zooniverse/Panoptes-Front-End/blob/9461eb8f83b4b12e0b9bee501cbe8d535783e1a7/app/redux/ducks/translations.js#L5-L11

Comment (camallen) [reply app/classifier/components/Intervention/Intervention.spec.js]: I commented out the two `counterpart.setFallbackLocale();` functions and it still had the fallback set. I was a bit lost at that point and decided `countepart` was magical

File: app/classifier/components/Intervention/Intervention.spec.js (hunk 4 above)
Comment (eatyourgreens) [inline app/classifier/components/Intervention/Intervention.spec.js]: By the way, is this split into multiple lines so that the linter doesn't complain? The concatenation is a bit strange.
